language/vm/src/normalized.rs

```diff
@@ -66,48 +67,69 @@ pub struct Struct {
 
 /// Normalized version of a `FunctionDefinition`. Not safe to compare without an associated
 /// `ModuleId` or `Module`.
-#[derive(Clone, Debug, Eq, PartialEq)]
+#[derive(Clone, Debug, Hash, Eq, PartialEq)]
 pub struct FunctionSignature {
     pub name: Identifier,
     pub type_parameters: Vec<AbilitySet>,
     pub formals: Vec<Type>,
     pub ret: Vec<Type>,
 }
 
+/// Normalized version of a `FriendDeclaration`.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Friend {
```

Comment:
Redundant? Seems like we could just use `Vec<ModuleId>` below.

Reply:
Removed.